src/include/esp-idf/bindings.h

```diff
@@ -265,9 +265,18 @@
 #include "esp_adc/adc_continuous.h"
 #endif
 #include "driver/twai.h"
+
 #if defined(CONFIG_IDF_TARGET_ESP32) || defined(CONFIG_IDF_TARGET_ESP32S2)
+#if ESP_IDF_VERSION_MAJOR > 5 ||                                               \
+    ESP_IDF_VERSION_MAJOR == 5 && ESP_IDF_VERSION_MINOR >= 1
+#include "driver/dac_continuous.h"
+#include "driver/dac_cosine.h"
+#include "driver/dac_oneshot.h"
+#else
```

Comment:
Why `#else` and not including both?

Reply:
I don't think the new drivers are available until 5.1, so the old driver should be left in for <=5.0 builds (if I understand the context correctly here).

Reply:
What I'm saying is: what if somebody is using the old driver **with** ESP IDF 5.1+? With your change, that won't be possible. Hence, for ESP IDF >= 5.1 we need to include both, and for earlier versions - only the old one.

Reply:
Ah, as it is deprecated, not removed. Made the change.

Reply:
I am concerned about [this line](https://github.com/espressif/esp-idf/blob/e3e112e6122dfd064a78f3dece8cdce17a472e13/docs/en/api-reference/peripherals/dac.rst?plain=1#L44), in particular, `the legacy driver cannot coexist with the new driver.`

Reply:
Define coexistence? As long as coexistence means user should not instantiate both drivers at runtime, we should be fine.